### ISSUE TITLE
Lighting Performance Regression versus 1.3b91

### ISSUE THREAD

**Issue by mlangsdorf** [JamzTheMan/MapTool#45]
Maps with large background images and large numbers (like 40+) of light sources are reasonably usable in 1.3b91. There is a slight lag when moving tokens but it's less than a few seconds.

The same map, loaded into 1.4.4, is completely unusable. There is a delay of 40-90 seconds between dragging a token and the map being updated to reflect the movement.

Steps to reproduce:
1. Load https://www.prismnet.com/~mlangsdo/eastern-theatre.cmpgn in MapTools 1.3.b91.
2. Go to the "Eastern Campaign" map and start dragging tokens. Notice a 1-3 second delay between releasing the token and the map being updated.
3. Load the same campaign in MapTool 1.4.4.
4. Go to the "Eastern Campaign" map and start dragging tokens. Notice a 60-120 second delay.

**Comment by JamzTheMan** [comment]
Thanks for that map/campaign! Something bad indeed is going on here!  I'll dig into this asap.

**Comment by Syndaryl** [comment]
I am subscribing to this issue as I'm also having big trouble with it. My map is four large (page sized) PNGs with black for the walls, transparent for the not-walls, and I used the token auto VBL feature to set up my vbl one click. Well, five clicks but still.

When it was one PNG, it was OK.
When I placed a second, it got worse.
When I placed the third and fourth it got impossible.

Turning vision to Day removes the problem, but I was rather appreciating that my players couldn't see the big picture, and didn't realize how big the space they were in really was. Once I set it to Day so we could play they kinda flipped out :)

**Comment by JamzTheMan** [comment]
@Syndaryl Your issue may be separate as you are using "new" functionality, namely TokenVBL. TokenVBL is more intense as it has to calculate it more often since it can "move".

Try transferring it to the normal VBL layer using a macro like:

```
[r: transferVBL(1)]
[r: finalVblData = "{'shape':'none'}"]
[r: setTokenVBL(finalVblData)] 
```

FYI: You can transfer VBL to the token via from the map layer:
```
[r: transferVBL(0)]
```

Also, Auto-Generated VBL *can* be very poor performing if two detailed. Like, if you manually drew around every single pixel, it would also be bad. If it's basically just straight walls, auto-generation should be ok.

FYI: You can drag/drop files right into the "issue" here so feel free to supply the campaign file for testing.

**Comment by JamzTheMan** [comment]
I made some improvements to Lights for the next release. This campaign file should be usable under 1.4.5.0 (just tested it) and I'll continue to look into this. There may still be room for additional improvements.

Note: You may want to clear FoW to see the best improvements. The fix involves changing lights from a true circle to a polygon which is 10-100 times faster to calculate the intersection with existing FoW. But if existing FoW already has "curves", performance may be poor. YMMV.

**Comment by mlangsdorf** [comment]
is there a beta build of 1.4.5 that I could test?

**Comment by JamzTheMan** [comment]
I just created one for Windows/Mac.  Appveyor isn't uploading the deb package for linux for some reason and looking into it...

https://github.com/JamzTheMan/MapTool/releases/tag/1.4.5.00-beta1

**Comment by JamzTheMan** [comment]
Should be fixed as of 1.4.5.3.
